cmd/kops/create_secret_cacert.go

```diff
@@ -0,0 +1,142 @@
+	in state store.`))
+
+	create_secret_cacert_example = templates.Examples(i18n.T(`
+	# Add a intermediate ca certificate key private key. and store the key
+	kops create secret ca-cert \
+		--cert ~/ca.pem \
+		--key ~/ca-key.pem \
+		--name k8s-cluster.example.com \
+		--state s3://example.com
+	`))
+
+	create_secret_cacert_short = i18n.T(`Add intermediate ca cert and private key.`)
+)
+
+type CreateSecretCaCertOptions struct {
+	ClusterName    string
+	PrivateKeyPath string
+	CertPath       string
+}
+
+func NewCmdCreateSecretCaCert(f *util.Factory, out io.Writer) *cobra.Command {
+	options := &CreateSecretCaCertOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "ca-cert",
+		Short:   create_secret_cacert_short,
+		Long:    create_secret_cacert_long,
+		Example: create_secret_cacert_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := rootCommand.ProcessArgs(args)
+			if err != nil {
+				exitWithError(err)
+			}
+
+			options.ClusterName = rootCommand.ClusterName()
+
+			err = RunCreateSecretCaCert(f, os.Stdout, options)
+			if err != nil {
+				exitWithError(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&options.CertPath, "cert", options.CertPath, "Path to ca cert")
+	cmd.Flags().StringVar(&options.PrivateKeyPath, "key", options.PrivateKeyPath, "Path to ca cert private key")
+
+	return cmd
+}
+
+// RunCreateSecretCaCert adds a custom intermidiate ca certificate and a pricate key
+func RunCreateSecretCaCert(f *util.Factory, out io.Writer, options *CreateSecretCaCertOptions) error {
+	if options.CertPath == "" {
+		return fmt.Errorf("error cert provided")
+	}
+
+	if options.PrivateKeyPath == "" {
+		return fmt.Errorf("error no private key provided")
+	}
+
+	cluster, err := GetCluster(f, options.ClusterName)
+	if err != nil {
+		return fmt.Errorf("error getting cluster: %q: %v", options.ClusterName, err)
+	}
+
+	clientSet, err := f.Clientset()
+	if err != nil {
+		return fmt.Errorf("error getting clientset: %v", err)
+	}
+
+	keyStore, err := clientSet.KeyStore(cluster)
+	if err != nil {
+		return fmt.Errorf("error getting keystore: %v", err)
+	}
+
+	options.CertPath = utils.ExpandPath(options.CertPath)
+	options.PrivateKeyPath = utils.ExpandPath(options.PrivateKeyPath)
+
+	certBytes, err := ioutil.ReadFile(options.CertPath)
+	if err != nil {
+		return fmt.Errorf("error reading user provided cert %q: %v", options.CertPath, err)
+	}
+	privateKeyBytes, err := ioutil.ReadFile(options.PrivateKeyPath)
+	if err != nil {
+		return fmt.Errorf("error reading user provided private key %q: %v", options.PrivateKeyPath, err)
+	}
+
+	privateKey, err := pki.ParsePEMPrivateKey(privateKeyBytes)
+	if err != nil {
+		return fmt.Errorf("error loading private key %q: %v", privateKeyBytes, err)
+	}
+	cert, err := pki.ParsePEMCertificate(certBytes)
+	if err != nil {
+		return fmt.Errorf("error loading certificate %q: %v", options.CertPath, err)
+	}
+
+	err = keyStore.StoreKeypair(fi.CertificateId_CA, cert, privateKey)
```

Comment:
Is this an intermediate ca then? ... as is appear your replacing the CA

Reply:
in this context it's actually a CA cert. but it can be a intermediate cert we use internally. But to stay with the context of copy it should be called CA not intermediate